src/lib/ImageUpload/ImageUpload.tsx

```diff
@@ -0,0 +1,245 @@
+            }
+        }
+    }
+
+    /**
+     * onDrop event handler in order to support drag & drop functionality.
+     * Notifies when the dragged item has been released over the target view.
+     * 
+     * @param e - Drag event information.
+     */
+    const onDrop = (e: DragEvent<HTMLInputElement>) => {
+        e.preventDefault();
+        setDragActive(false);
+        setError(undefined);
+        const acceptedFiles: File[] = [];
+        const declinedFiles: File[] = [];
+        const acceptedPreviews: string[] = [];
+        const dt = new DataTransfer();
+
+        if (e.dataTransfer.files) {
+            for (const file of Object.values(e.dataTransfer.files)) {
+                    const valid = isValidFile(file, acceptedFiles, declinedFiles, acceptedPreviews);
+                    if (valid) {
+                        dt.items.add(file);
+                    }
+            }
+        }
+
+        if (onImageSelected) {
+            onImageSelected(acceptedFiles[0]);
+        }
+        setUploadedImages(acceptedFiles);
+        setPreviews(acceptedPreviews);
+        if (inputRef.current) {
+            inputRef.current.files = dt.files.length ? dt.files : null;
+        }
+
+        if (declinedFiles.length > 0) {
+            setError(`${declinedFiles.length > 1
+                ? `${declinedFiles.length} files`
+                : 'File'} could not be loaded. Check for the correct file type and size (max size: ${maxSize}MB)`);
+        }
+    }
+
+    /**
+     * onDragOver event handler in order to support drag & drop functionality.
+     * Notifies when the dragged item has been dragged over the target view.
+     * 
+     * @param e - Drag event information
+     */
+    const onDragOver = (e: DragEvent<HTMLInputElement>) => {
+        e.preventDefault();
+        
+        if (!dragActive) {
+            setDragActive(true);
+        }
+    }
+
+    /**
+     * onDragLeave event handler in order to support drag & drop functionality.
+     * Notifies when the dragged item has left the target view.
+     * 
+     * @param e - Drag event information
+     */
+    const onDragLeave = (e: DragEvent<HTMLInputElement>) => {
+        if (dragActive) {
+            setDragActive(false);
+        }
+    }
+
+    useEffect(() => {
+        if (hasUploaded) {
+            clearInput();
+        }
+    }, [hasUploaded])
+
+    return <div className="c4imgupload--wrapper">
+        {previews && previews.length > 0 && <div className="c4imgupload--preview">
+            <div className="preview">
+                <img
+                    className="img"
+                    alt="Uploaded file preview"
+                    src={previews[0]}
+                />
+                <div className="cutout">
+                    <div className="left" />
+                    <div className="center" />
+                    <div className="right" />
+                </div>
```

Comment:
Future feature idea (not now!), would be cool to have an optional prop for a 'screen ratio' of how the uploaded image might be cropped or shown on the frontend. 

ie- currently this cutout is square i think (for avatars which makes sense) but we could remove that opinion or just make it default and make this adaptable for use cases where the end-use of the image is landscape or uncropped etc

Reply:
I'll add this as an issue \O/